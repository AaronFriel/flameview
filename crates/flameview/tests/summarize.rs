#[test]
fn summary_limits_lines_and_coverage() {
    let data = include_str!("../../../tests/data/perl.txt");
    let tree = flameview::loader::collapsed::load(data.as_bytes()).unwrap();
    let out = tree.summarize(5, 0.90);
    let n = out.lines().count();
    assert!(n <= 6, "root + \u{2264}5 children");
    assert!(out.contains("(root)"));
    assert!(out.contains("  ")); // at least one indented line
    insta::assert_snapshot!(out, @r#"(root) (100.0%, 1000)
  foo (60.0%, 600)
    bar (40.0%, 400)
  baz (35.0%, 350)"#);
}
