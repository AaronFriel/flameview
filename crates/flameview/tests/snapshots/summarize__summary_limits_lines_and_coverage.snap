---
source: crates/flameview/tests/summarize.rs
expression: out
---
(root) (100.0%, 34)
  java (79.4%, 27)
    start_thread (79.4%, 27)
      java_start (79.4%, 27)
        JavaThread::run (79.4%, 27)
          JavaThread::thread_main_inner (79.4%, 27)
